packages/vaadin-dialog/src/vaadin-dialog.js

```diff
@@ -311,26 +278,8 @@ class DialogElement extends ThemePropertyMixin(
     this.$.overlay.addEventListener('vaadin-overlay-outside-click', this._handleOutsideClick.bind(this));
     this.$.overlay.addEventListener('vaadin-overlay-escape-press', this._handleEscPress.bind(this));
 
-    this._observer = new FlattenedNodesObserver(this, (info) => {
-      this._setTemplateFromNodes(info.addedNodes);
-    });
-  }
-
-  /**
-   * @param {!Array<!Node>} nodes
-   * @protected
-   */
-  _setTemplateFromNodes(nodes) {
-    this._contentTemplate =
-      nodes.filter((node) => node.localName && node.localName === 'template')[0] || this._contentTemplate;
-  }
-
-  /** @private */
-  _removeNewRendererOrTemplate(template, oldTemplate, renderer, oldRenderer) {
-    if (template !== oldTemplate) {
-      this._contentTemplate = undefined;
-    } else if (renderer !== oldRenderer) {
-      this.renderer = undefined;
+    if (window.Vaadin.templateRendererCallback) {
```

Comment:
Safer to check for `window.Vaadin` namespace first.

```suggestion
    if (window.Vaadin && window.Vaadin.templateRendererCallback) {
```

Reply:
Yeah, thanks! Fixed.

Reply:
I wish someday we could use the `.?` operator.